package.json

```diff
@@ -43,9 +44,15 @@
     "lint": "esw --color src --ext .ts",
     "fix": "yarn lint --fix",
     "eslint-check": "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
-    "build": "tsc",
+    "build": "rm -rf lib/* && tsc && tsc -p tsconfig.esm.json",
     "watch": "tsc --watch"
   },
+  "sideEffects": false,
+  "exports": {
```

Comment:
The `exports` field is redundant with `main` and `module`. You don't really need it here. It is supported with NodeJS 12+.